### PR TITLE
webcord-vencord: override electron to use electron_24

### LINKED ISSUE #255729: webcord-vencord: segfault on run
### Describe the bug
After recently updating NixOS, webcord-vencord regularly segfaults upon running it. Normal webcord does not appear to have the same problem.

### Steps To Reproduce
1. Install webcord-vencord
2. Run webcord-vencord in a terminal with `webcord`
3. A blank window appears for a moment and then disappears. The terminal output will say it had a segfault.

### Expected behavior
I expected it to not segfault and run normally

### Screenshots
N/A

### Additional context
I have a flake-based setup
My dotfiles: https://github.com/Anomalocaridid/dotfiles

I have tried deleting the data in `~/.config/WebCord,` and it sort of worked, but then after closing and re-opening webcord-vencord once, it resumed crashing with a segfault.

Output when I run webcord-vencord in a terminal:
```
[WebSocket] Listening at port 6463.
[1]    274720 segmentation fault (core dumped)  webcord
```

### Notify maintainers
@FlafyDev
@NotAShelf 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.3-xanmod1, NixOS, 23.11 (Tapir), 23.11.20230915.ace5093`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - nixpkgs: `not found`
```

### PR DESCRIPTION
**Pull request by LudovicoPiero**
## Description of changes

Latest Webcord update changed to downgrade Electron to version 25, which unfortunately resulted in a segfault when launching Webcord-Vencord on Wayland. Therefore, we decided to revert the downgrade for now. It's possible that this issue will be resolved in Electron version 27

Fixes #255729 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NotAShelf** [comment]
Looks alright, but why the electron_25 = electron_24? I think it makes more sense to do something like `electron ? electron_24` as we seem to keep tinkering with the electron version.

**Review by NotAShelf — COMMENTED** [review]
Looks good. 

Though, is this closed by #256158?

**Comment by LudovicoPiero** [comment]
> Looks good.
> 
> Though, is this closed by #256158?

Nope
```console
[nix-shell:~/.cache/nixpkgs-review/pr-256158]$ ./results/webcord-vencord/bin/webcord --enable-features=UseOzonePlatform --ozone-platform=wayland --enable-features=WebRTCPipeWireCapturer --enable-wayland-ime
[50768:0920/160713.285294:FATAL:gpu_init.cc(495)] Passthrough is not supported, GL is egl, ANGLE is
[WebSocket] ポート6464にてListenしています。
[アップデート] アプリケーションは最新です！
[GPU:133] crashed
[50825:0920/160714.049269:FATAL:gpu_init.cc(495)] Passthrough is not supported, GL is egl, ANGLE is
[GPU:133] crashed
[50848:0920/160714.666973:FATAL:gpu_init.cc(495)] Passthrough is not supported, GL is egl, ANGLE is
[GPU:133] crashed
```

**Comment by NotAShelf** [comment]
Welp. Can't win 'em all.

**Review by NotAShelf — APPROVED** [review]

File: pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix
Comment (NotAShelf) [inline pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:5]: this probably can be replaced with `electron ? electron_24` in order to get rid of the redundant let in

Comment (FlafyDev) [reply pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:5]: I don't really understand

doing  `electron ? electron_24,` results in variable `electron_24` not found.
and either way it wouldn't care about the `? electron_24` part because callPackage will assign electron to electron_26 (cuz the packacge already exists)

I think we should keep it as `electron_25 = electron_24` for now. The only problem with it that I can think of is that when webcord changes electron versions, that line will no longer do anything. But no idea how to solve it and this probably shouldn't be dealt with in this PR..

Comment (NotAShelf) [reply pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:5]: correction, pkgs.electron_24*

Comment (LudovicoPiero) [reply pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:5]: Ugghh, Got this error using `pkgs.electron_24`
```console
       error: undefined variable 'pkgs'

       at /home/ludovico/Code/nixpkgs/pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:5:14:

            4| , vencord-web-extension
            5| , electron ? pkgs.electron_24
             |              ^
            6| }:
```

I think it's just better to use the old 
```nix
electron_25 = electron_24
```

Comment (nevivurn) [reply pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:5]: Why don't we do something like what java is doing with jre? Let the package accept `electron` and set `electron = electron_xx;` in callPackage in `top-level/all-packages.nix`.

For webcord-vencord, we could do `callPackage ... { webcord = webcord.override { electron = electron_xx; }; }`

Comment (FlafyDev) [reply pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/default.nix:5]: Good idea. That will be better than just doing `electron_25 = electron_24` if Webcord maintainers are alright with it. But I will approve the changes anyway(after removing the let in) since the point of this PR is to just override the electron version to something else like in previous commits.

**Review by FlafyDev — APPROVED** [review]

**Review by NotAShelf — APPROVED** [review]

**Comment by NotAShelf** [comment]
Guess this can be merged, but we should be cleaning up the derivation instead.

**Comment by yu-re-ka** [comment]
I believe wayland should work on electron_26

**Comment by LudovicoPiero** [comment]
> I believe wayland should work on electron_26

IIRC, electron_26 has some weird grey screen bug. For example, #254539.

**Comment by eclairevoyant** [comment]
electron 27 is now available fwiw

**Pull request closed without merging**
